cogs/embeds.py

```diff
@@ -1,24 +1,142 @@
+            if img_url.endswith(">"): img_url = img_url[:-1]
+            try:
+                embed.set_image(url=img_url)
+            except:
+                print("Error setting image")
+
+        async def handle_fields(embed: discord.Embed):
+            while True:
+                await ctx.send("You are now adding another field. Please use the following format: `\"Title\" \"Value\"`. You can type `done` to complete the embed.")
+                field_msg = await self.bot.wait_for("message", check=check, timeout=timeout)
+                if is_done(field_msg):
+                    break
+                [title, value] = await handle_field(field_msg)
+                title = title[:256]
+                value = value[:1024]
+                embed.add_field(name=title, value=value)
+
+        async def handle_field(field_msg: discord.Message):
+            output = []
+            current = None
+            field_tokens = re.split(" +", field_msg.content.strip())
+
+            for token in field_tokens:
+                if token.startswith("\"") and current is None:
+                    current = ""
+                
+                if current is not None:
+                    current = current + token + " "
+                
+                if token.endswith("\"") and current is not None:
+                    index = len(token) - 2
+                    if index < 0 or token[index] != "\\":
+                        data = current.strip()
+                        data = data[1:-1]
+                        output.append(data)
+                        current = None
+
+            return output
+
+        async def handle_channel():
+            while True:
+                try:
+                    channel_msg: discord.Message = await self.bot.wait_for("message", check=check, timeout=timeout)
+                    if len(channel_msg.channel_mentions) is 0:
+                        await ctx.send("Could not find that channel")
+                    else:
+                        channel = channel_msg.channel_mentions[0]
+                        return channel
+                except TimeoutError:
+                    await ctx.send("Reached Timeout. Discarding Embed")
+                    break
+
+        async def handle_replacement_msg(embed: discord.Embed, channel: discord.TextChannel):
+            msg_id_msg: discord.Message = await self.bot.wait_for("message", check=check, timeout=timeout)
+            msg_id = msg_id_msg.content.strip()
+            try:
+                msg = await channel.fetch_message(msg_id)
+                await msg.edit(embed=embed)
+            except:
+                await channel.send(embed=embed)
+
+        timeout = 60 * 1000
+        embed = discord.Embed()
+        
+        await ctx.send("Please type the description of the embed. Use `skip` to skip this.")
+        desc: discord.Message = await self.bot.wait_for("message", check=check, timeout=timeout)
```

Comment:
When using `wait_for`, use the try-except statement like shown in the [documentation](https://discordpy.readthedocs.io/en/stable/ext/commands/api.html?highlight=wait_for#discord.ext.commands.Bot.wait_for).
```py
try:
    desc: discord.Message = await self.bot.wait_for("message", check=check, timeout=timeout)
except asyncio.TimeoutError:
    # handle when command times out
```
On its own, the code raises a TimeoutError and doesn't execute the rest of the function.